session.go

```diff
@@ -119,8 +119,8 @@ type session struct {
 	processInfo atomic.Value
 	txn         kv.Transaction // current transaction
 	txnFuture   *txnFuture
-	// goCtx is used for cancelling the execution of current transaction.
-	goCtx      goctx.Context
+
+	// cancelFunc is used for cancelling the execution of current transaction.
 	cancelFunc goctx.CancelFunc
```

Comment:
The `cancelFunc` may be accessed by another goroutine. Need to be protected by mutex.